CMakeLists.txt

```diff
@@ -405,12 +406,26 @@ else()
   set(HAVE_SELINUX 0)
 endif()
 
+cmake_push_check_state()
+set(CMAKE_REQUIRED_FLAGS "-Werror=nonnull")
```

Comment:
What if the compiler doesn't support `-Werror=nonnull` but happily compiles the code?

Reply:
Good question. Quick test on my machine (gcc 13.1.1 emits an error, clang 16.0.3 emits a warning):
```
$ gcc -Werror=nonnu2ll test.c; echo $?
cc1: error: ‘-Werror=nonnu2ll’: no option ‘-Wnonnu2ll’; did you mean ‘-Wnonnull’?
1
$ clang -Werror=nonnu2ll test.c; echo $?
warning: unknown warning option '-Werror=nonnu2ll'; did you mean '-Werror=nonnull'? [-Wunknown-warning-option]
1 warning generated.
0
```

This happens with your previous version (both emit a warning):
```
$ gcc test.c; echo $?
test.c: In function ‘main’:
test.c:4:38: warning: unknown option after ‘#pragma GCC diagnostic’ kind [-Wpragmas]
    4 |         #pragma GCC diagnostic error "-Wnonnu2ll"
      |                                      ^~~~~~~~~~~~
test.c:4:38: note: did you mean ‘-Wnonnull’?
0
$ clang test.c; echo $?
test.c:4:31: warning: unknown warning group '-Wnonnu2ll', ignored [-Wunknown-warning-option]
        #pragma GCC diagnostic error "-Wnonnu2ll"
                                     ^
1 warning generated.
0
```

Is this the intended behavior? Or are you just curious what happens?

Reply:
`nonun2ll`?

Reply:
To see what happens if the warning is unknown, I picked a name that is guaranteed to be unknown ;)

Reply:
That's not what I meant. E.g. CompCert would fail the test now because it cannot handle `-Werror=nonnull` as `CMAKE_REQUIRED_FLAGS`. This is semantically different from `FSTATAT_PATH_ACCEPTS_NULL`.

Reply:
Ah, I see. So should I go back to setting the `#pragma` instead of  `CMAKE_REQUIRED_FLAGS`?

Reply:
Both versions have (dis)advantages but I'd prefer the `#pragma` one as it works for the two main compilers and is just meant to suppress warnings. :man_shrugging:

Reply:
I was not aware of this difference in semantics, thanks for catching that! I changed it back to the `#pragma` as I agree it fits the purpose of suppressing a warning better.